### PR TITLE
🌱 github: add workflow to auto-approve golangci-lint if ok-to-test label is set

### LINKED ISSUE #9215: Run golangci-lint as soon as ok-to-test is set
Today we have significant toil for maintainers because they always have to manually approve the golangci-lint action for outside contributors (by clicking on approve in the "checks box").

Goal of this issue is to ensure the action is run automatically as soon as the ok-to-test label is applied.

Proposed solution:
1. Configure in the repo settings that actions only require approval for first time GitHub users
2. Update the action to immediately fail with a nice error message if the ok-to-test label is missing
  * Prior art: `contains(github.event.pull_request.labels.*.name, 'ok-to-test')`

Probably needs a bit of trial-and-error on a fork, but I think this should work

For 1.
image

There is potentially another alternative, which is approve workflows automatically if ok-to-test is set: https://github.com/kubernetes-sigs/metrics-server/blob/19802fd979b3c8cdf23297cebe9dc62448e9dc9d/.github/workflows/gh-workflow-approve.yaml#L14. This seems somewhat okay as it only requires actions write permissions. But I think I would prefer the option I mentioned above

### PR DESCRIPTION
**Pull request by chrischdi**
**What this PR does / why we need it**:

Prior art: https://github.com/kubernetes-sigs/metrics-server/blob/9dbb59e99addce5956f65d6e44749fb36eef339e/.github/workflows/gh-workflow-approve.yaml

Tested at https://github.com/chrischdi/cluster-api-provider-vsphere/pull/4

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #9215

### CODE REVIEW

File: .github/workflows/pr-gh-workflow-approve.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+name: Approve GH Workflows
```
Comment (sbueringer) [inline .github/workflows/pr-gh-workflow-approve.yaml:1]: please add it to jobs.md

**Review by sbueringer — COMMENTED** [review]
lgtm overall, just one nit

**Comment by sbueringer** [comment]
/area ci

/lgtm

Let's give others some time to review and otherwise merge next week and just give it a try

**Review by killianmuldoon — COMMENTED** [review]
/lgtm

**Pull request merged**